### PR TITLE
fix: Implement auto installation of uvloop on linux arm

### LINKED ISSUE #395: [BUG] `uvloop` ModuleNotFoundError: No module named 'uvloop' on Ubuntu Docker Image
**Description**

Running a boilerplate robyn project within an ubuntu docker image on mac m1 fails due to missing `uvloop` runtime dependency. Able to build the image but an error occurs at runtime. 

`uvloop` is a [conditional](https://github.com/sansyrox/robyn/blob/main/pyproject.toml#L16) dependency which does not get installated in this environment.

**Expected Behavior**

I would expect that, if `uvloop` is required in my environment that it is installed via `pip install robyn`. This seems like a bug in the conditional install logic? 

**Screenshots**

```
docker run 618feeec50d8                             
INFO:robyn:Starting server at 0.0.0.0:1001 
INFO:robyn:Press Ctrl + C to stop 

Process Process-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/multiprocess/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.10/dist-packages/multiprocess/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.10/dist-packages/robyn/processpool.py", line 54, in spawn_process
    loop = initialize_event_loop()
  File "/usr/local/lib/python3.10/dist-packages/robyn/processpool.py", line 21, in initialize_event_loop
    import uvloop
ModuleNotFoundError: No module named 'uvloop'
```

**Environment:**

```
├── Dockerfile
└── src
    └── foo
        ├── __init__.py
        └── app.py
```

```
FROM ubuntu:22.04

WORKDIR /workspace

RUN apt-get update -y && \
    apt-get install -y python3.10 python3-pip
    
RUN pip install --no-cache-dir --upgrade robyn

# sansyrox/robyn:pyproject.toml conditional python dependencies not installed :(
# to fix:
# RUN pip install uvloop>=0.17.0


COPY ./src /workspace/

EXPOSE 1001

CMD ["python3.10", "/workspace/foo/app.py", "--log-level=DEBUG"]
```

- Host OS:
```
uname -a
Darwin My-MBP 21.6.0 Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:35 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T8101 arm64
```

- Within the image:
```
docker run --rm -it --entrypoint bash  618feeec50d8 
root@0827d3192a02:/workspace# uname -a
Linux 0827d3192a02 5.10.76-linuxkit #1 SMP PREEMPT Mon Nov 8 11:22:26 UTC 2021 aarch64 aarch64 aarch64 GNU/Linux
root@0827
```

---

[Optional] **Additional Context**

```
cat ./src/unus/app.py 
from robyn import Robyn

app = Robyn(__file__)

@app.get("/")
async def h(request):
    return "Hello, world!"

app.start(port=1001, url="0.0.0.0")
```

---

### PR DESCRIPTION
**Pull request by sansyrox**
**Description**

This PR fixes #395

fix: Implement auto installation of uvloop on linux arm

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies = [
   'nestd==0.3.1',
 # conditional
   'uvloop == 0.17.0; sys_platform == "darwin"',
+  'uvloop == 0.17.0; sys_platform == "linux"',
```
Comment (AntoineRR) [inline pyproject.toml:18]: Can we do `sys_platform != "win32"`?
Example [here](https://packaging.python.org/en/latest/specifications/core-metadata/#requires-external-multiple-use)

Comment (sansyrox) [reply pyproject.toml:18]: @AntoineRR , just `win32` will not work. Windows has arm support now and even `x86_64` possibly. Not sure how windows architecture is mentioned

Comment (AntoineRR) [reply pyproject.toml:18]: I think you are confusing several things, `arm` and `x86_64` are CPU architectures, while windows is the OS. `sys_platform` calls `sys.platform` which gets specification of the OS : https://docs.python.org/3/library/sys.html#sys.platform

Comment (sansyrox) [reply pyproject.toml:18]: > while windows is the OS. sys_platform calls sys.platform which gets specification of the OS :

Oh, thanks @AntoineRR . TIL 😄 

I will update the PR

File: pyproject.toml [hunk 2]
```diff
@@ -14,7 +14,7 @@ dependencies = [
   'multiprocess == 0.70.14',
   'nestd==0.3.1',
 # conditional
-  'uvloop == 0.17.0; sys_platform == "darwin"',
+  'uvloop == 0.17.0; sys_platform != "win32"',
   'uvloop == 0.17.0; platform_machine == "x86_64"',
   'uvloop == 0.17.0; platform_machine == "i686"'
```
[the remark this answers is not in the document]
Comment (sansyrox) [reply pyproject.toml:19]: @AntoineRR , I remember from our previous tests that it was needed.

I can try again

[the remark this answers is not in the document]
Comment (AntoineRR) [reply pyproject.toml]: Ok, we'll restore it if it's needed in the end

**Review by AntoineRR — APPROVED** [review]

**Pull request merged**